yb-voyager/cmd/analyzeSchema.go

```diff
@@ -402,6 +417,148 @@ func checkStmtsUsingParser(sqlInfoArr []sqlInfo, fpath string, objType string) {
+				}
+			}
+			colName := column.GetColumnDef().GetColname()
+			if len(column.GetColumnDef().GetTypeName().GetArrayBounds()) > 0 {
+				//For Array types and storing the type as "array" as of now we can enhance the to have specific type e.g. INT4ARRAY
+				_, ok := tableToColumnUnsupportedDataType[fullyQualifiedName]
+				if !ok {
+					tableToColumnUnsupportedDataType[fullyQualifiedName] = make(map[string]string)
+				}
+				tableToColumnUnsupportedDataType[fullyQualifiedName][colName] = "array"
+			} else if slices.Contains(unsupportedIndexDatatypes, typeNameOnIdx0) || slices.Contains(unsupportedIndexDatatypes, typeNameOnIdx1) {
+				_, ok := tableToColumnUnsupportedDataType[fullyQualifiedName]
+				if !ok {
+					tableToColumnUnsupportedDataType[fullyQualifiedName] = make(map[string]string)
+				}
+				if slices.Contains(unsupportedIndexDatatypes, typeNameOnIdx0) {
+					tableToColumnUnsupportedDataType[fullyQualifiedName][colName] = typeNameOnIdx0
+				} else if slices.Contains(unsupportedIndexDatatypes, typeNameOnIdx1) {
+					tableToColumnUnsupportedDataType[fullyQualifiedName][colName] = typeNameOnIdx1
+				}
+			}
+		}
+	}
+}
+
+func reportUnsupportedIndexesOnComplexDatatypes(createIndexNode *pg_query.Node_IndexStmt, sqlStmtInfo sqlInfo, fpath string) {
+	indexName := createIndexNode.IndexStmt.GetIdxname()
+	relName := createIndexNode.IndexStmt.GetRelation()
+	schemaName := relName.GetSchemaname()
+	tableName := relName.GetRelname()
+	fullyQualifiedName := lo.Ternary(schemaName != "", schemaName+"."+tableName, tableName)
+	displayObjName := fmt.Sprintf("%s ON %s", indexName, fullyQualifiedName)
+	/*
+		e.g.
+		1. CREATE INDEX tsvector_idx ON public.documents  (title_tsvector, id);
+		stmt:{index_stmt:{idxname:"tsvector_idx"  relation:{schemaname:"public"  relname:"documents"  inh:true  relpersistence:"p"  location:510}  access_method:"btree"
+		index_params:{index_elem:{name:"title_tsvector"  ordering:SORTBY_DEFAULT  nulls_ordering:SORTBY_NULLS_DEFAULT}}  index_params:{index_elem:{name:"id"
+		ordering:SORTBY_DEFAULT  nulls_ordering:SORTBY_NULLS_DEFAULT}}}}  stmt_location:479  stmt_len:69
+
+		2. CREATE INDEX idx_json ON public.test_json ((data::jsonb));
+		stmt:{index_stmt:{idxname:"idx_json"  relation:{schemaname:"public"  relname:"test_json"  inh:true  relpersistence:"p"  location:703}  access_method:"btree"
+		index_params:{index_elem:{expr:{type_cast:{arg:{column_ref:{fields:{string:{sval:"data"}}  location:722}}  type_name:{names:{string:{sval:"jsonb"}}  typemod:-1
+		location:728}  location:726}}  ordering:SORTBY_DEFAULT  nulls_ordering:SORTBY_NULLS_DEFAULT}}}}  stmt_location:676  stmt_len:59
+	*/
+	if createIndexNode.IndexStmt.AccessMethod != "btree" {
+		return // Right now not reporting any other access method issues with such types.
+	}
+	_, ok := tableToColumnUnsupportedDataType[fullyQualifiedName]
+	if ok {
+		for _, param := range createIndexNode.IndexStmt.GetIndexParams() {
```

Comment:
Yes, it will work for multi-column index cases as well we will report the first column we find in the list as an unsupported one. I have a test case in analyze for this